api/handler/v1beta1/resource.go

```diff
@@ -74,36 +77,84 @@ func (sv *RuntimeServiceServer) ReadResource(ctx context.Context, req *pb.ReadRe
 	}, nil
 }
 
-func (sv *RuntimeServiceServer) DeployResourceSpecification(req *pb.DeployResourceSpecificationRequest, respStream pb.RuntimeService_DeployResourceSpecificationServer) error {
+func (sv *RuntimeServiceServer) DeployResourceSpecification(stream pb.RuntimeService_DeployResourceSpecificationServer) error {
 	startTime := time.Now()
+	errNamespaces := []string{}
 
-	namespaceSpec, err := sv.namespaceService.Get(respStream.Context(), req.GetProjectName(), req.GetNamespaceName())
-	if err != nil {
-		return mapToGRPCErr(sv.l, err, "unable to get namespace")
-	}
-	var resourceSpecs []models.ResourceSpec
-	for _, resourceProto := range req.GetResources() {
-		adapted, err := sv.adapter.FromResourceProto(resourceProto, req.DatastoreName)
+	for {
+		request, err := stream.Recv()
 		if err != nil {
-			return status.Errorf(codes.Internal, "%s: cannot adapt resource %s", err.Error(), resourceProto.GetName())
+			if err == io.EOF {
+				break
+			}
+			stream.Send(&pb.DeployResourceSpecificationResponse{
+				Success: false,
+				Ack:     true,
+				Message: err.Error(),
+			})
+			return err // immediate error returned (grpc error level)
+		}
+		namespaceSpec, err := sv.namespaceService.Get(stream.Context(), request.GetProjectName(), request.GetNamespaceName())
+		if err != nil {
+			stream.Send(&pb.DeployResourceSpecificationResponse{
+				Success: false,
+				Ack:     true,
+				Message: err.Error(),
+			})
+			errNamespaces = append(errNamespaces, request.NamespaceName)
+			continue
+		}
+		var resourceSpecs []models.ResourceSpec
+		var errAdaptedResource error
+		for _, resourceProto := range request.GetResources() {
+			adapted, err := sv.adapter.FromResourceProto(resourceProto, request.DatastoreName)
+			if err != nil {
+				sv.l.Error(fmt.Sprintf("%s: cannot adapt resource %s", err.Error(), resourceProto.GetName()))
+				errAdaptedResource = errors.New("resource adapt is failed")
+				continue
```

Comment:
This should be using `break` as we are checking the error below for this loop and using the same error for any failure.

Can be done as part of a separate cleanup pr.

Reply:
updated the implementation, though still using the `continue` instead of break because we want to accumulate the error messages.